src/hotspot/share/runtime/os.hpp

```diff
@@ -336,6 +336,9 @@ class os: AllStatic {
   // than "free" memory (`MemFree` in `/proc/meminfo`) because Linux can free memory
   // aggressively (e.g. clear caches) so that it becomes available.
   static julong available_memory();
+#ifdef LINUX
+  static julong used_memory();
+#endif
```

Comment:
I don't understand. Now it's a linux only API? Wasn't the intention to have it available on all, but a shared impl for all except Linux, like in `os.cpp`?

Something like:
```
LINUX_ONLY(static julong pd_used_memory();)
...
julong os::used_memory() {
#ifdef LINUX
  return os::pd_used_memory();
#else
  return os::physical_memory() - os::available_memory();
#endif
}

Reply:
> I don't understand. Now it's a linux only API? Wasn't the intention to have it available on all, but a shared impl for all except Linux, like in `os.cpp`?
> 
> 
> 
> Something like:
> 
> ```
> 
> LINUX_ONLY(static julong pd_used_memory();)
> 
> ...
> 
> julong os::used_memory() {
> 
> #ifdef LINUX
> 
>   return os::pd_used_memory();
> 
> #else
> 
>   return os::physical_memory() - os::available_memory();
> 
> #endif
> 
> }

That was indeed the intention.

Reply:
Yeah that's way better, d'oh. I was too happy just getting to delete code.